### PR TITLE
ACL refactor

### PR DESCRIPTION
**Pull request by kradalby**
General QoL improvements to the ACL code, naming, splitting up larger functions, some more tests at different layers.

Rewrite of the filtering of peers served as part of MapResponses, fixing #1372, undoing #1239, #1377, redoing #320.

Small contribution to laying the groundwork for improving SSH support.

### CODE REVIEW

File: acls.go [hunk 1]
```diff
@@ -820,8 +682,255 @@ func expandGroup(
...
+
+type Match struct {
+	Srcs  []netip.Prefix
+	Dests []netip.Prefix
+}
+
+func MatchFromFilterRule(rule tailcfg.FilterRule) Match {
+	match := Match{
+		Srcs:  []netip.Prefix{},
+		Dests: []netip.Prefix{},
```
Comment (kradalby) [inline acls.go:896]: Replace with [IPSet](https://pkg.go.dev/go4.org/netipx#IPSetBuilder)

**Comment by ventsislav-georgiev** [comment]
@kradalby Is there any chance of an image pushed to a public registry for trying out the changes directly?

**Comment by kradalby** [comment]
@ventsislav-georgiev sorry, its being continuously updated, so its not really feasible with our current release pipeline, we dont support or build docker officially. There is a binary attached to the build job on this PR.

**Comment by loprima-l** [comment]
Hi, about the potential ACL API, what is the current plan ?

**Comment by kradalby** [comment]
@loprima-l no plan, I have no intention of implementing an API for ACLs.

The current state is that some people want it, and have discussed how to do it, and we will accept an implementation that fits into the project. But it is currently not part of our priorities, and I do not think this will change for a long time.

**Review by juanfont — APPROVED** [review]

**Pull request merged**